### ISSUE TITLE
`node-param-fixed-collection-type-unsorted-items`: rule throws a TypeError, so ESLint exits 2 and reports no findings at all

### ISSUE THREAD

**Issue by romerjon** [ivov/eslint-plugin-n8n-nodes-base#217]
`node-param-fixed-collection-type-unsorted-items` throws a `TypeError` instead of reporting, which makes ESLint exit 2 with **no findings at all**.

The rule is an AST heuristic over *any* object literal carrying `displayName` / `name` / `values`. Handed an object that is not an n8n node parameter — a plain domain data structure — it walks a fixed number of `.parent` hops, guards only that the ancestor **exists**, and then calls `.properties.find(…)` on a node that has no `.properties`.

The consequence is worse than a false positive: the rule **throws** rather than reporting, so ESLint exits **rc=2** with a stack trace and prints **no `problems` summary line whatsoever**. Every other rule's result for that run is lost. On a codebase where this construct exists, the whole lint step becomes unreadable rather than merely noisy.

## Environment

| | |
|---|---|
| `eslint-plugin-n8n-nodes-base` | **1.16.5** |
| ESLint | **9.39.3** (flat config) |
| Node | 22.23.1 |
| `NODE_ENV` | **unset** (relevant to #197) |
| Parser | `@typescript-eslint/parser` |

Minimal flat config used for every reproduction below — the plugin is registered directly, with **one rule armed at a time**, so nothing else can be responsible:

```js
// eslint.config.mjs
import tsParser from "@typescript-eslint/parser";
import n8nNodesBase from "eslint-plugin-n8n-nodes-base";

export default [
  {
    files: ["**/*.ts"],
    languageOptions: { parser: tsParser, ecmaVersion: 2022, sourceType: "module" },
    plugins: { "n8n-nodes-base": n8nNodesBase },
    rules: { "n8n-nodes-base/RULE_NAME": "error" },
  },
];
```

The file under test is `nodes/Demo/Demo.node.ts`. Every snippet below is synthetic and self-contained — you can paste it and run it as-is.

I noticed #196 is still open on ESLint v9 support. These all reproduce on 9.39.3 with flat config; if there is a supported version matrix you would rather I verify against, tell me and I will re-run them.

## Reproduction

```ts
// nodes/Demo/Demo.node.ts
// A domain data structure. Its elements carry displayName / name / description / values, but it is
// NOT an n8n node parameter — it is a contract table consumed by the node's own code.
export const SEGMENTATION_CONTRACT = {
  filters: {
    displayName: "Filters",
    name: "filters",
    description: "Supported segmentation filters",
    values: [
      { displayName: "Zebra", name: "zebra", description: "z" },
      { displayName: "Alpha", name: "alpha", description: "a" },
    ],
  },
};
```

## Observed

```
Oops! Something went wrong! :(

ESLint: 9.39.3

TypeError: Cannot read properties of undefined (reading 'find')
Occurred while linting .../nodes/Demo/Demo.node.ts:4
Rule: "n8n-nodes-base/node-param-fixed-collection-type-unsorted-items"
    at getStringProperty (node_modules/eslint-plugin-n8n-nodes-base/dist/lib/ast/getters/nodeParameter.getters.js:48:38)
    at Object.getDisplayName (node_modules/eslint-plugin-n8n-nodes-base/dist/lib/ast/getters/nodeParameter.getters.js:57:10)
    at isAddressFixedCollectionSection (node_modules/eslint-plugin-n8n-nodes-base/dist/lib/rules/node-param-fixed-collection-type-unsorted-items.js:84:58)
    at ObjectExpression (node_modules/eslint-plugin-n8n-nodes-base/dist/lib/rules/node-param-fixed-collection-type-unsorted-items.js:49:13)
```

`eslint` exit code: **2**. Zero `✖ N problems` lines.

## Expected

The rule reports nothing for a subject it cannot resolve. A rule that cannot identify its subject should be silent, never throw.

## Suggested fix

A type guard on the ancestor chain in `isAddressFixedCollectionSection` — confirm the resolved ancestor actually has `.properties` before calling `.find` on it, rather than only confirming the ancestor is not `undefined`.

## Note — this is a class, not an instance

Other rules share the same heuristic and mis-fire on the same kind of non-parameter object; `node-param-description-identical-to-display-name` is one. Any rule that keys on "an object with `displayName` and `description`" will see plain data modules whenever the configured glob covers a whole package rather than only description files. A guard in the shared getter would cover the family.

Possibly related but distinct: #111 is about `displayOptions` in this rule's autofix; this report is about the rule throwing before it reports anything.
